odml/validation.py

```diff
@@ -4,12 +4,49 @@
 """
 
 import re
+
+from enum import Enum
+
 from . import dtypes
 
 LABEL_ERROR = 'error'
 LABEL_WARNING = 'warning'
 
 
+class ValidationID(Enum):
```

Comment:
One thing that keeps coming to mind while reading through this but I'm not sure of is whether it makes more sense to identify errors and warnings instead of the handlers.  The behaviour right now would be the same, but would change the name of the class/enum to something like IssueID.  Conceptually the difference is that the handler (a function that validates something) isn't what's being identified but instead it's the issue of the file or object being validated.  This is also reinforced when I see that the ID is used to construct a ValidationError object.

I'm thinking of this in a similar way to how code linters show errors or warnings that have specific codes.  The code identifies the issue.  This change might also be necessary if at some point we consider a handler (or a user writes one) that can produce more than one type of error or warning.

I'm not entirely 100% sure on this, so I'm just posting it as something to consider.

Reply:
I like `IssueID`!